### DISCUSSION TITLE
how to run TFT_eSPI on Wrover kit v4.1?

### DISCUSSION THREAD

**Discussion by baxino** [Bodmer/TFT_eSPI#982] [General]
Can u helpme ? Iam using a Platformio  and iam trying to run whatever on lcd screen , for that discussion let assume that I wanna run colour_test example. compiling is ok no error and.... nothing on lcd. 
I tryed that code and nothing....
`
#include <SPI.h>

#define ILI9341_DRIVER

#define TFT_DC 0
#define TFT_CS 15
#define TFT_RST 2
#define TFT_CLK 18
#define TFT_MOSI 23
#define TFT_MISO 19
#define TFT_BACKLIGHT 5

#include <TFT_eSPI.h>       // Hardware-specific library

TFT_eSPI tft = TFT_eSPI();  // Invoke custom library

void setup(void) {
  tft.init();

  tft.fillScreen(TFT_BLACK);
  
  // Set "cursor" at top left corner of display (0,0) and select font 4
  tft.setCursor(0, 0, 4);

  // Set the font colour to be white with a black background
  tft.setTextColor(TFT_WHITE, TFT_BLACK);

  // We can now plot text on screen using the "print" class
  tft.println("Intialised default\n");
  tft.println("White text");
  
  tft.setTextColor(TFT_RED, TFT_BLACK);
  tft.println("Red text");
  
  tft.setTextColor(TFT_GREEN, TFT_BLACK);
  tft.println("Green text");
  
  tft.setTextColor(TFT_BLUE, TFT_BLACK);
  tft.println("Blue text");

  delay(5000);
`
but when i tryed pure adafruit library with modyfication  named "WROVER_KIT_LCD"  from there 
https://github.com/espressif/WROVER_KIT_LCD/blob/master/examples/graphicstest/graphicstest.ino
IT RUN!!! holy moly... but it doesnt have all feature like drawString itp.

How the hell, i can run that library (tft_eSPI by bodmer) on wrover kit ?:O

**Comment by Bodmer** [comment]
You need to configure the settings in a configuration file, [see here](https://github.com/Bodmer/TFT_eSPI/tree/master/Tools/PlatformIO).

The simplest solution if you are not familiar with the PlatformIO environment is to use the Arduino IDE.

**Reply by khaoulaMOS** [reply]
I AM USING ARDUINO AND STILL  NOT WORKING  I CONFIGURE THE FILE FROM HERE https://docs.espressif.com/projects/esp-idf/en/stable/esp32/hw-reference/esp32/get-started-wrover-kit.html#get-started-esp-wrover-kit-v4-1-lcd-connector

**Reply by khaoulaMOS** [reply]
![image](https://user-images.githubusercontent.com/83791953/143428090-d0fc2bd8-de11-4624-add0-f87ef934d4f1.png)

**Comment by khaoulaMOS** [comment]
it works for me when i add 
#define TFT_MISO 25
#define TFT_MOSI 23
#define TFT_SCLK 19
#define TFT_CS   22  // Chip select control pin
#define TFT_DC    21  // Data Command control pin
#define TFT_RST   18 // Invoke custom library
#define TFT_BL 5 
in arduino code and i add in void setup   pinMode(TFT_BL, OUTPUT);

**Reply by Bodmer** [reply]
The Arduino IDE hides any #define in the .ino sketch from the libraries, so all defines used by the library must be in the setup file local to the library. Things are a little different in PlatformIO, so in that case the .ini file can be used:
https://github.com/Bodmer/TFT_eSPI/blob/master/Tools/PlatformIO/Configuring%20options.txt
